inference-engine/src/mkldnn_plugin/memory_descs/cpu_memory_desc_utils.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cpu_memory_desc.h"
+#include "memory_descs/cpu_memory_desc_utils.h"
+#include "mkldnn_memory.h"
+#include "memory_descs/onednn_blocked_memory_desc.h"
+#include "utils/general_utils.h"
+#include "utils/cpu_utils.hpp"
+#include <limits>
+#include <vector>
+#include <numeric>
+#include <blob_factory.hpp>
+#include <dnnl_types.h>
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+namespace MKLDNNPlugin {
+
+MemoryDescPtr MemoryDescUtils::makeDescriptor(const mkldnn::memory::desc &desc) {
+    if (desc.data.format_kind == dnnl_blocked) {
+        return std::unique_ptr<OnednnBlockedMemoryDesc>(new OnednnBlockedMemoryDesc(desc));
+    } else {
+        return MKLDNNPlugin::make_unique<MKLDNNMemoryDesc>(desc);
+    }
+}
+
+std::unique_ptr<MKLDNNMemoryDesc> MemoryDescUtils::convertToMKLDNNMemoryDesc(const MemoryDesc& desc) {
+    if (MemoryDescType::CpuBlocked == desc.getType()) {
+        return convertToMKLDNNMemoryDesc(*(desc.as<CpuBlockedMemoryDesc>()));
+    } else if (MemoryDescType::Mkldnn & desc.getType()) {
+        return MKLDNNPlugin::make_unique<MKLDNNMemoryDesc>(desc.as<MKLDNNMemoryDesc>()->getMklDesc());
```

Comment:
replaced with clone + downcast

> It seems that cloning already MKLDNNMemoryDesc dynamic object is excessive operation.

at this moment i have no idea how we can fix it, because we need to return ```unique_ptr```

Reply:
Let's try `return MKLDNNPlugin::make_unique<MKLDNNMemoryDesc>(*desc.as<MKLDNNMemoryDesc>());` then.